### ISSUE TITLE
Add flow loop API enhancement

### ISSUE THREAD

**Issue by ginty** [Origen-SDK/origen_testers#46]
Add the ability for a flow to define a loop section like this:

~~~ruby
# Run these tests 100 times
flow.loop 100 do
  test :blah1
  test :blah2
end
~~~

On testers that support flow loops it should use the available features, if not it should generate a flattened inline version.

Expect that it will be implemented like this for the current crop of testers:

V93K - For loop
UltraFLEX - Generate the contained block as a separate flow, then invoke it multiple times via "call"
J750 - Generated it flat and inline

**Comment by info-rchitect** [comment]
@ginty any ETA on this for the V93K?  This became a must have for one of our products as the for loop is based on a variable that can be updated based on test results.

**Comment by ginty** [comment]
No plans until now, will try and make time for it soon.

Still waiting for @arkohler to push the test cases to allow this one to be investigated and fixed first: https://github.com/Origen-SDK/atp/pull/7

**Comment by info-rchitect** [comment]
@ginty here isan example of the for loop:

~~~shell
   for @vol = 0.3; @vol < 0.4 ; @vol = @vol + 0.1; do
       test_number_loop_increment = 0
       {
          svlr_level_command(14,1,"VDDC",0.7 + @vol);
          {
             run_and_branch(Fmax_Search_g1)
             then
             {
             }
             else
             {
             }
          }, open,"fmax_search", ""
       } 
~~~

**Comment by info-rchitect** [comment]
@ginty any lick here?  As Origen is growing here, this feature request has been ticking up significantly.

**Comment by ginty** [comment]
No progress on this so far.

Looking at your example @info-rchitect, it seems to go beyond the original intent of providing a simple loop API.

What do envisage as the application-facing API to generate something like that?

**Comment by info-rchitect** [comment]
@ginty yeah i am trying to see what parts of this just need to be in Ruby and what, if any , should be in V93k.  stay tuned...

**Comment by info-rchitect** [comment]
Bumping this issue as the [topic came up on stack overflow](https://stackoverflow.com/questions/61613338/origen-testers-atp-flow-loop-method-use-case).  There is a loop ATP API but it is not hooked up to the V93K generator. I will try to make a PR.

**Comment by redxeth** [comment]
Well I wouldn't say it fully closes it-- it will help for SMT7.  If others can help me with the syntax for STM8 and UF then I can add that as well to this PR.

**Comment by redxeth** [comment]
Reopened it since currently only resolved for v93k SMT7.   Leaving open in case anyone wants to do for SMT8, Ultraflex, etc.
